### ISSUE TITLE
comparison="And" in StatusEffects sometimes fails when it shouldn't

### ISSUE THREAD

**Issue by 3e849f2e5c** [Regalis11/Barotrauma#4631]
Reported by Udrakan#0750 in #baro-modding

Example:
The conditionals always fail because StatusEffect.HasRequiredConditions() targets include both the Holdable ItemComponent and the Item itself so the code first checks `<Conditional condition="lt 100"/>` against the Item which may fail or succeed but then proceeds to check it against the Holdable component which will always fail since Holdable has no property matching "condition" causing it to fail the entire StatusEffect
```XML
<Item name="ANDRod" identifier="androd" Tags="smallitem" cargocontaineridentifier="metalcrate" scale="0.5" description="">
    <Sprite texture="Content/Items/Tools/tools.png" depth="0.55" sourcerect="410,1,19,68" />
    <Body radius="6" height="55" density="20" />
    <Holdable holdpos="30,-15" handle1="0,0" slots="RightHand,LeftHand" msg="ItemMsgPickUpSelect" >
        <StatusEffect type="OnContained" target="This" stackable="false" disabledeltatime="true" delay="0.5" comparison="And">
            <Conditional condition="lt 100"/>
            <Conditional hastag="smallitem"/>
            <Fire size="1"/>
        </StatusEffect>
    </Holdable>
</Item>
```

**Comment by itchyOwl** [comment]
Have to fix this to fix the oxygen tank shelves refilling welding fuel.

**Comment by itchyOwl** [comment]
Should be fixed in Regalis11/Barotrauma-development@02678a0.

**Comment by Rokvach** [comment]
Confirmed working, closing.
